internal/controller/nonadminbackup_controller.go

```diff
@@ -195,45 +225,57 @@ func (r *NonAdminBackupReconciler) validateSpec(ctx context.Context, logger logr
 //	ctx: Context for the request.
 //	logger: Logger instance for logging messages.
 //	nab: Pointer to the NonAdminBackup object.
-func (r *NonAdminBackupReconciler) syncVeleroBackupWithNonAdminBackup(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
-	veleroBackupName := function.GenerateVeleroBackupName(nab.Namespace, nab.Name)
-	if veleroBackupName == constant.EmptyString {
-		return false, errors.New("unable to generate Velero Backup name")
+func (r *NonAdminBackupReconciler) createVeleroBackupAndSyncWithNonAdminBackup(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	if nab.Status.VeleroBackup == nil || nab.Status.VeleroBackup.NameUUID == constant.EmptyString {
+		return false, errors.New("unable to get Velero Backup UUID from NonAdminBackup Status")
 	}
 
-	veleroBackup := velerov1.Backup{}
-	veleroBackupLogger := logger.WithValues("VeleroBackup", types.NamespacedName{Name: veleroBackupName, Namespace: r.OADPNamespace})
-	err := r.Get(ctx, client.ObjectKey{Namespace: r.OADPNamespace, Name: veleroBackupName}, &veleroBackup)
+	veleroBackupNameUUID := nab.Status.VeleroBackup.NameUUID
+
+	veleroBackup, err := function.GetVeleroBackupByLabel(ctx, r.Client, r.OADPNamespace, veleroBackupNameUUID)
+
 	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			veleroBackupLogger.Error(err, "Unable to fetch VeleroBackup")
-			return false, err
-		}
-		// Create VeleroBackup
-		veleroBackupLogger.Info("VeleroBackup not found")
+		// Case in which more then one VeleroBackup is found with the same label UUID
+		logger.Error(err, "Failed to find single VeleroBackup object", "UUID", veleroBackupNameUUID)
+		return false, err
+	}
+
+	if veleroBackup == nil {
+		logger.Info("VeleroBackup with label not found, creating one", "UUID", veleroBackupNameUUID)
 
 		backupSpec := nab.Spec.BackupSpec.DeepCopy()
 		backupSpec.IncludedNamespaces = []string{nab.Namespace}
 
-		veleroBackup = velerov1.Backup{
+		veleroBackup := velerov1.Backup{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:        veleroBackupName,
+				Name:        veleroBackupNameUUID,
 				Namespace:   r.OADPNamespace,
 				Labels:      function.GetNonAdminLabels(),
 				Annotations: function.GetNonAdminBackupAnnotations(nab.ObjectMeta),
 			},
 			Spec: *backupSpec,
 		}
 
+		// Add NonAdminBackup's veleroBackupNameUUID as the label to the VeleroBackup object
+		// We don't add this as an argument of GetNonAdminLabels(), because there may be
+		// situations where NAC object do not require NabOriginUUIDLabel
+		veleroBackup.Labels[constant.NabOriginNameUUIDLabel] = veleroBackupNameUUID
```

Comment:
this does not make sense to me, it is duplication of information (we can get this info from the name of the object, that is exactly the same)

Reply:
This is intended to improve usability for users, primarily administrators.

Reply:
yes labels help in managing and filtering objects.

Reply:
`GetNonAdminLabels()` was meant to be valid for every NAC object also the one which may not require `veleroBackupNameUUID`, that's why I did not want to encorporate this in the function. We can have similar to annotations function:

```
function.GetNonAdminBackupLabels(nab.ObjectMeta)
```

But it will just have that single line of code and we will need to have new function for that, so I felt it's unnecessary addition.